inflation/lp/InflationLP.py

```diff
@@ -0,0 +1,1890 @@
+    @property
+    def sparse_extra_inequalities(self) -> coo_matrix:
+        """Extra inequalities in sparse matrix form."""
+        ineq_row, ineq_col, ineq_data = [], [], []
+        nof_inequalities = len(self.extra_inequalities)
+        for row_idx, ineq in enumerate(self.extra_inequalities):
+            nof_vars = len(ineq)
+            ineq_row.extend(np.repeat(row_idx, nof_vars))
+            ineq_col.extend([self.compmonomial_to_idx[x] for x in ineq])
+            ineq_data.extend(ineq.values())
+        return coo_matrix((ineq_data, (ineq_row, ineq_col)),
+                          shape=(nof_inequalities, self.n_columns))
+
+    @property
+    def sparse_inequalities(self) -> coo_matrix:
+        """All inequalities (minimal and extra) in sparse matrix form."""
+        return vstack((self.minimal_sparse_inequalities,
+                       self.sparse_extra_inequalities))
+
+    def _coo_vec_to_mon_dict(self,
+                             col: np.ndarray,
+                             data: np.ndarray) -> Dict:
+        """Convert a COO vector to a dictionary of monomials.
+        
+        Parameters
+        ----------
+        col : numpy.ndarray
+            Column indices of the COO vector.
+        data : numpy.ndarray
+            Data of the COO vector.
+            
+        Returns
+        -------
+        Dict
+            Dictionary of monomials.
+        """
+        return dict(zip(self.monomials[col].flat, data))
+
+    def _coo_vec_to_name_dict(self, 
+                              col: np.ndarray,
+                              data: np.ndarray) -> Dict:
+        """Convert a COO vector to a dictionary of monomial names.
+        
+        Parameters
+        ----------
+        col : numpy.ndarray
+            Column indices of the COO vector.
+        data : numpy.ndarray
+            Data of the COO vector.
+        
+        Returns
+        -------
+        Dict
+            Dictionary of monomial names.
+        """
+        return dict(zip(self.monomial_names[col].flat, data))
+
+    def _coo_mat_to_dict(self,
+                         input_coo_mat: coo_matrix,
+                         string_keys: bool = False) -> List[Dict]:
+        """Convert a COO matrix to a list of dictionaries.
+        
+        Parameters
+        ----------
+        input_coo_mat : coo_matrix
+            Input COO matrix.
+        string_keys : bool, optional
+            Whether to use string keys or not, by default, ``False``.
+        
+        Returns
+        -------
+        List[Dict]
+            List of dictionaries.
+        """
+        input_lil_mat = input_coo_mat.tolil(copy=False)
+        args_iter = zip(input_lil_mat.rows, input_lil_mat.data)
+        if string_keys:
+            return [self._coo_vec_to_name_dict(*args) for args in args_iter]
+        else:
+            return [self._coo_vec_to_mon_dict(*args) for args in args_iter]
+
+    def _mon_dict_to_coo_vec(self, monomials_dict: Dict) -> coo_matrix:
```

Comment:
This function is still being used, confirm that we are not deprecating it in this merge?

Reply:
See sparse_objective, sparse_known_vars, sparse_lowerbounds, sparse_upperbounds

Reply:
We are not deprecating this function yet, so it's ok.